apps/condo/domains/common/components/PostMessageProvider/globalHandlers.ts

```diff
@@ -138,4 +141,34 @@ export const useUpdateProgressBarHandler: () => RequestHandler<'CondoWebAppUpdat
         return { updated: true }
         // TODO(DOMA-5171): Adding miniAppTaskUIInterface in deps causing rerender hell!
     }, [userId])
+}
+
+export const useRedirectHandler: () => RequestHandler<'CondoWebAppRedirect'> = () => {
+    const router = useRouter()
+
+    return useCallback(({ url, target = '_self' }) => {
+        let success = false
+        if (!isSafeUrl(url)) {
+            throw new Error('The URL is not safe')
+        }
+
+        if (typeof window !== 'undefined') {
+            if (target === '_blank') {
+                window.open(url, target)
+                success = true
+            } else if (target === '_self') {
+                const urlOrigin = extractOrigin(url)
```

Comment:
what if I'll pass something like `/contacts`, will it work?

Reply:
Yes, it's return `null`

Reply:
But it's allowed url, isn't it?

Reply:
Yes, in this case no error will be thrown